easybuild/framework/easyconfig/tools.py

```diff
@@ -276,17 +276,21 @@ def get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR, robot_path=None):
 
 def alt_easyconfig_paths(tmpdir, tweaked_ecs=False, from_pr=False):
     """Obtain alternative paths for easyconfig files."""
-    # path where tweaked easyconfigs will be placed
-    tweaked_ecs_path = None
+
+    # paths where tweaked easyconfigs will be placed, easyconfigs listed on the command line take priority, tweaked
+    # dependencies are also created but these will only be appended to the robot path (and only used if strictly
+    # necessary)
+    tweaked_ecs_paths = None
     if tweaked_ecs:
-        tweaked_ecs_path = os.path.join(tmpdir, 'tweaked_easyconfigs')
+        tweaked_ecs_paths = {'tweaked_ecs_path': os.path.join(tmpdir, 'tweaked_easyconfigs'),
+                             'tweaked_ecs_deps_path': os.path.join(tmpdir, 'tweaked_dep_easyconfigs')}
```

Comment:
Using `tweaked_ecs_paths[0]`/`target_dirs[0]` and `tweaked_ecs_paths[1]`/`target_dirs[1]` is fairly clear to me (`[0]` for path to prepend, `[1]` for path to append)

Maybe you can also split it up like this to make it more clear:

```python
tweaked_ecs_paths, tweaked_ecs_deps_path = target_dirs
```

That's even better, since it leaves no question how large the `target_dirs` tuple is, and it makes it clear it's not a list of directories as the name suggests.

The problem with a dictionary is that you could update it somewhere along the way to change the elements or add more etc., which is not the intention here. A tuple is immutable (although, admittedly, the list values *in* the tuple can be updated themselves).